lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -247,17 +247,18 @@ def request(  # noqa
             OpenSearchHttpError if hosts are unreachable
         """
 
-        def call(url: str) -> requests.Response:
+        def call(urls: List[str]) -> requests.Response:
             """Performs an HTTP request."""
             for attempt in Retrying(
                 retry=retry_if_exception_type(requests.RequestException)
                 | retry_if_exception_type(urllib3.exceptions.HTTPError),
                 stop=stop_after_attempt(retries),
                 wait=wait_fixed(1),
-                before_sleep=error_http_retry_log(logger, retries, method, url, payload),
```

Comment:
WIP: removed for now, I think we can replace the `url` for `urls` here and log all the options available...

Reply:
PS.: this should be a debug log level: if we got to the point of asking ourselves which retry is failing, then something is going wrong...

The actual, final error, should then be a `error` level as it does prevent the functioning of the charm.